src/server/zset_family.cc

```diff
@@ -1716,14 +1724,23 @@ void ZSetFamily::ZAdd(CmdArgList args, ConnectionContext* cntx) {
       return (*cntx)->SendError(kScoreNaN);
     }
     string_view member = ArgS(args, i + 1);
-    auto [_, inserted] = members_set.insert(member);
-    unique_members &= inserted;
+    if (unique_members && num_members > 2) {
+      auto [_, inserted] = members_set.insert(member);
+      unique_members &= inserted;
+    }
     members.emplace_back(val, member);
   }
   DCHECK(cntx->transaction);
 
   if (unique_members) {
-    std::sort(members.begin(), members.end());
+    if (num_members == 2) {  // fix unique_members for this special case.
```

Comment:
I suggest that we cover in a unit test for 2 and 3 members cases.

Reply:
done